Libraries/config/Sources/Config.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <config/Config.h>
+#include <plist/plist.h>
+
+#include <memory>
+#include <vector>
+
+using config::Config;
+
+template<typename T>
+T _loadValue(plist::Object *config, std::string const &configKey);
+
+std::vector<std::string> _loadValue(plist::Object *config, std::string const &configKey)
+{
+    std::vector<std::string> values;
+
+    if (config == nullptr) {
+        return values;
+    }
+
+    auto extract = plist::Extract(config, configKey);
+    plist::Array *array;
+    if (extract != nullptr && (array = plist::CastTo<plist::Array>(extract))) {
+        for (auto iter = array->begin(); iter != array->end(); iter++) {
+            auto value = plist::CastTo<plist::String>(iter->get());
+
+            if (value != nullptr) {
+                values.push_back(value->value());
+            }
+        }
+    }
+
+    return values;
+}
+
+Config::Config(libutil::Filesystem const *filesystem, std::string const &defaultConfigPath)
+{
+    if (char const *environConfigPath = getenv("CONFIG_PATH")) {
+        _path = std::string(environConfigPath);
```

Comment:
It's a little weird to have this override the parameter based on global state. If there was a `static` function on `Config` to get the default path that would then be passed into the load method, then the two operations could be separated out. That would let clients pick exactly what behavior they want.